### PR TITLE
Update launcher to support uploading logs (infolog.txt) to S3 or Gist

### PR DESCRIPTION
**Pull request by Born2Crawl**

### CODE REVIEW

File: src/log_uploader.js [hunk 1]
```diff
@@ -4,48 +4,85 @@ const { dialog, clipboard } = require('electron');
...
+	'6939487543',
+	'386B497255',
+	'673149596A',
+	'48692B6C6A',
+	'33786D734D',
+	'434A5A6D66',
+	'4A4F415875',
+	'7A5543574A'
 ];
-const github_access_token = sec.join('');
```
Comment (badosu) [inline src/log_uploader.js:20]: Might want to export these to environment variables or something. I don't mind as it is atm.

Just keep in mind your logs will be mine 😈

Comment (Born2Crawl) [reply src/log_uploader.js:20]: Yeah, and Gists will be mine... There's really no place we can store it atm, launcher needs to know them when running. So either expose them in config, or fetch from some server

File: src/spring_log.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ log.transports.console = (msg) => {
 	}
 };
 
-const logPath = `${springPlatform.writePath}/spring-launcher.log`;
+const logPath = `${springPlatform.writePath}/infolog.txt`; // Who needs spring-launcher.log anyway?
```
Comment (gajop) [inline src/spring_log.js:24]: won't be accepting this change

File: src/log_uploader.js [hunk 3]
```diff
@@ -59,24 +96,61 @@ function upload_ask() {
...
+
+	gistClient.setToken(github_access_token)
+
+	return new Promise((resolve, reject) => {
+		gistClient.create(content).then(newGist => {
+			resolve({ 'url': newGist?.html_url });
+		}).catch(err => {
+			reject(err.toString());
+		});
 	});
```
Comment (gajop) [inline src/log_uploader.js:150]: Why would we change from GH repo to gists?

File: package.json [hunk 4]
```diff
@@ -44,16 +44,17 @@
...
     "electron-log": "^4.3.1",
     "electron-settings": "^3.2.0",
     "electron-updater": "^4.3.5",
     "extract-zip": "^2.0.1",
+    "gist-client": "^1.1.1",
     "got": "^11.8.2",
     "node-7z": "^2.1.2",
-    "octonode": "^0.10.2",
-    "sdfz-demo-parser": "^4.1.2",
+    "sdfz-demo-parser": "^4.6.0",
```
Comment (gajop) [inline package.json:57]: can't just add random changes like this.

**Review by gajop — CHANGES_REQUESTED** [review]

**Comment by gajop** [comment]
think I might as well close this as I won't be going in this direction;
log upload has been fixed.

File: src/spring_log.js (hunk 2 above)
Comment (Born2Crawl) [reply src/spring_log.js:24]: When Spring crashes, launcher logs don't contain the stacktrace, so we have to ask users to find and upload the infolog.txt manually. So from this perspective, launcher logs are much less useful - that was the intent behind this change.

File: package.json (hunk 4 above)
Comment (Born2Crawl) [reply package.json:57]: I think it was introduced by 'npm audit fix' to resolve some serious vulnerability it was complaining about.

File: src/spring_log.js (hunk 2 above)
Comment (gajop) [reply src/spring_log.js:24]: Works for me: https://logs.springrts.com/logfiles/6974/ , what's the actual problem? 
Do they restart/close the launcher before trying to upload or what?

File: package.json (hunk 4 above)
Comment (gajop) [reply package.json:57]: I think it was introduced by including various other changes from bar/spring-launcher, not necessarily something you did

File: src/log_uploader.js (hunk 3 above)
Comment (Born2Crawl) [reply src/log_uploader.js:150]: Issues are public, Gists are private and are only visible when the URL is known. Logs might contain users; personal information, so it's best to give them control over whom to share it with.
Also, creating an issue for every log upload is too much noise in the repository...

File: src/spring_log.js (hunk 2 above)
Comment (Born2Crawl) [reply src/spring_log.js:24]: No, it's something else. Here's a recent example, you can compare the launcher log and infolog: https://discord.com/channels/549281623154229250/549282926596980738/915083659965517844
(sorry, don't want to attach someone's logs in here)

Comment (gajop) [reply src/spring_log.js:24]: The link you sent has two logs, in which spring-launcher's log contains everything that infolog has and then a bit more. It's the infolog.txt that gets cut short, although neither of the two have a stacktrace. 

image

File: src/log_uploader.js (hunk 3 above)
Comment (gajop) [reply src/log_uploader.js:150]: Repos can be private, although that would make the posted issue invisible even to the person posting. So gists do make sense in that regard.

Regarding noise, you can just use a special repo for that. It's why I made https://github.com/beyond-all-reason/Logs/issues originally. The benefit of a custom repo is organization. Devs can easily search and categorize logs if they're posted in one repo.

**Pull request closed without merging**